src/app/venter/detailed-complaint/detailed-complaint.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, Params } from '@angular/router';
+import { IComplaint, IComplaintComment, IPostComment } from '../../interfaces';
+import { DataService } from '../../data.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { API } from '../../../api';
+import { MatDialog } from '@angular/material';
+import { EditCommentComponent } from '../edit-comment/edit-comment.component';
+import { VenterDataService } from '../venter-data.service';
+
+@Component({
+  selector: 'app-detailed-complaint',
+  templateUrl: './detailed-complaint.component.html',
+  styleUrls: ['./detailed-complaint.component.css']
+})
+export class DetailedComplaintComponent implements OnInit {
+  detailedComplaint: IComplaint;
+  public complaintId: string;
+  public comment: IPostComment;
+  public userid: string;
+  public SubscribeToComplaint: string;
+  public selectedindex = 0;
+  public editedComment: IPostComment;
+
+  constructor(public dataService: DataService,
+    public router: Router,
+    public activatedRoute: ActivatedRoute,
+    public snackBar: MatSnackBar,
+    public editCommentDialog: MatDialog,
+    public venterDataService: VenterDataService,
+  ) { }
+
+  ngOnInit() {
+    this.comment = {} as IPostComment;
+    this.editedComment = {} as IPostComment;
+    /* Get profile if the user is logged in */
+    if (this.dataService.isLoggedIn()) {
+      this.getUser();
```

Comment:
This is still being treated as sync.

Reply:
Just looked through this carefully; if I'm right that `userid` isn't being used anywhere besides checking if the comment was made by the user, you could let this be the way it is and explicitly initialize `userid` to null to be sure